src/Symfony/Bridge/Doctrine/Form/ChoiceList/DoctrineChoiceLoader.php

```diff
@@ -146,7 +146,7 @@ public function loadChoicesForValues(array $values, $value = null)
 
         // Optimize performance in case we have an object loader and
         // a single-field identifier
-        if (null === $value && !$this->choiceList && $this->objectLoader && $this->idReader->isSingleId()) {
+        if (!($value instanceof \Closure) && !$this->choiceList && $this->objectLoader && $this->idReader->isSingleId()) {
```

Comment:
Actually, the new test doesn't pass with `!is_callable($value)`, which means that it breaks the optimization.

That's because the [default value](https://github.com/symfony/symfony/blob/v2.7.10/src/Symfony/Bridge/Doctrine/Form/Type/DoctrineType.php#L222) set by `DoctrineType` is callable, so we wouldn't perform the optim by default.

I'm not sure that I really understand *when* we want the optimization to be performed. One thing I'm sure of is that if no value is passed to `choice_value`, we want it. Based on this I tried the following
```php
$isDefaultValue = is_array($value)
    && is_callable($value)
    && $value[0] instanceof \Symfony\Bridge\Doctrine\Form\ChoiceList\IdReader
    && $value[1] === 'getIdValue';

if ($isDefaultValue && !$this->choiceList && $this->objectLoader && $this->idReader->isSingleId()) {
    // optimize !
}
```
Is there a better way to know that `$value` is actually the default value ? Or a better way to know that the optimization can be performed ?